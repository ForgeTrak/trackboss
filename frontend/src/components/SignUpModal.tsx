import React, { useEffect, useState } from 'react';
import {
    Modal,
    ModalOverlay,
    ModalContent,
    ModalCloseButton,
    Button,
    Divider,
    Heading,
    ModalFooter,
    SimpleGrid,

} from '@chakra-ui/react';

interface modalProps {
  isOpen: boolean,
  onClose: () => void,
  attendeesList: any[],
  familyMembers: any[],
}

function signUp(attendingFamily: any[], member: any) {
    // Call controller to sign up member, then once it confirms...
    return attendingFamily.concat(member);
}

export default function SignUpModal(props: modalProps) {
    const [attendingFamily, setAttendingFamily] = useState<any>([]);
    useEffect(() => {
        async function setAttendance() {
            props.familyMembers.forEach((member) => {
                // If the family member is attending the event, then add them to the list of attendingFamily
                // (make their button orange)
                if (props.attendeesList.some((attendee: any) => JSON.stringify(member) === JSON.stringify(attendee))) {
                    attendingFamily.push(member);
                }
            });
            setAttendingFamily(attendingFamily);
        }
        setAttendance();
    }, []);

    return (
        <Modal size="xl" isOpen={props.isOpen} onClose={props.onClose}>
            <ModalOverlay />
            <ModalContent>
                <Heading pl={2} pr={2} textAlign="left">Select members to sign up</Heading>
                <Divider mb={5} />
                <ModalCloseButton />
                <SimpleGrid spacing={2} columns={3}>
                    {
                        props.familyMembers.map((member) => (
                            <Button
                                key={member.member_id}
                                onClick={
                                    () => {
                                        setAttendingFamily(signUp(attendingFamily, member));
                                    }
                                }
                                m={3}
                                backgroundColor={
                                    attendingFamily.includes(member)
                                        ? 'orange.300' : 'grey.300'
                                }
                                _hover={{ bg: 'orange.100' }}
                            >
                                {member.name}
                            </Button>
                        ))
                    }
                </SimpleGrid>
                <Divider />
                <ModalFooter>
                    <Button
                        variant="ghost"
                        mr={3}
                        size="sm"
                        onClick={
                            () => {
                                props.onClose();
                            }
                        }
                    >
                        Close
                    </Button>
                </ModalFooter>
            </ModalContent>
        </Modal>
    );
}
