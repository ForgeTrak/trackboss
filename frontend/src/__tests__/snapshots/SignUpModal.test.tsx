import React from 'react';
import renderer from 'react-test-renderer';
import '@testing-library/jest-dom/extend-expect';
import SignUpModal from '../../components/SignUpModal';

function onClose() {
    // a void function
}

it('renders all props correctly', () => {
    const eventAttendees = [
        {
            name: 'testing',
        },
        {
            name: 'testing again',
        },
    ];
    const familyMembers = [
        {
            name: 'testing',
        },
        {
            name: 'testing again',
        },
    ];
    const modal = renderer.create(
        <SignUpModal
            isOpen
            onClose={onClose}
            attendeesList={eventAttendees}
            familyMembers={familyMembers}
        />,
    ).toJSON();
    expect(modal).toMatchSnapshot();
});
