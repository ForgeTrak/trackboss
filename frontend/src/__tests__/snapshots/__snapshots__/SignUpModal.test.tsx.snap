// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders all props correctly 1`] = `<span />`;

exports[`sign up modal renders all props correctly 1`] = `<span />`;
